app/ante/eth.go

```diff
@@ -190,6 +184,19 @@ func (egcd EthGasConsumeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simula
 			gasWanted += txData.GetGas()
 		}
 
+		// user balance is already checked during CheckTx so there's no need to
+		// verify it again during ReCheckTx
+		if ctx.IsReCheckTx() {
+			// Use new context with gasWanted = 0
+			// Otherwise, there's an error on txmempool.postCheck (tendermint)
+			// that is not bubbled up. Thus, the Tx never runs on DeliverMode
+			// Error: "gas wanted -1 is negative"
+			// For more information, see issue #1554
+			// https://github.com/evmos/ethermint/issues/1554
+			gasWanted = uint64(0)
```

Comment:
```suggestion
```

don't need this because we don't skip the `gasWanted` logic, and it won't be `-1`.